app/src/main/java/ch/epfl/sdp/contamination/databaseIO/ConcreteCachingDataSender.java

```diff
@@ -30,12 +39,60 @@
 
     SortedMap<Date, Location> lastPositions;
     private GridFirestoreInteractor gridInteractor;
+    private StorageManager<Date,Location> positionHistory;
 
     public ConcreteCachingDataSender(GridFirestoreInteractor interactor) {
         this.gridInteractor = interactor;
         this.lastPositions = new TreeMap<>();
+        this.positionHistory = initStorageManager();
     }
+    private StorageManager<Date, Location> openStorageManager() {
+
+        return new ConcreteManager<Date, Location>(
+                CoronaGame.getContext(),
+                "last_positions.csv",
+                date_position -> {
+                    try {
+                        return CoronaGame.dateFormat.parse(date_position);
+                    } catch (ParseException e) {
+                        throw new IllegalArgumentException("The file specified has wrong format: field 'date_position'");
+                    }
+                }, location->{
+
+                return stringToLocation(location);
+                }
+        );
+    }
+    private Location stringToLocation(String s){
+        Pattern p = Pattern.compile("-?\\d+(,\\d+)*?\\.?\\d+?");
```

Comment:
I'm wondering what the effectiveness of this would be vs. using a simple _String.split(",")_?

Reply:
haha good point.

Reply:
I'm pretty conviced that we should use an actual csv lib whenever we deal with csv instead of rolling our own stuff but that's beyond this PR